components/Ink/src/private/MDCInkLayer.m

```diff
@@ -33,13 +41,25 @@
 
 @implementation MDCInkLayer
 
++ (BOOL)supportsSecureCoding {
+  return YES;
+}
+
+- (instancetype)init {
+  self = [super init];
+  if (self) {
+    _inkColor = [UIColor colorWithWhite:0 alpha:(CGFloat)0.08];
+  }
+  return self;
+}
+
 - (instancetype)initWithLayer:(id)layer {
   self = [super initWithLayer:layer];
   if (self) {
     _endAnimationDelay = 0;
     _finalRadius = 0;
     _initialRadius = 0;
-    _inkColor = [UIColor colorWithWhite:0 alpha:0.08f];
+    _inkColor = [UIColor colorWithWhite:0 alpha:(CGFloat)0.08];
```

Comment:
Curious why you decided to cast instead? Also I noticed you are using this number more than once, maybe worth putting as a const at the top.

Reply:
0.08f and 0.08 are not equivalent on 64-bit architectures. 0.08 can't be represented perfectly in binary, so the colors are different.

Reply:
ah this is interesting! Do we have other places where this could be a problem too? I am assuming we usually use the `f` approach

Reply:
We do have a lot of code that uses `1.f` or `0.3f` to suppress warnings. In general it's not a huge issue if the numbers aren't critical (watch out for floating point rounding errors!). In this case, it makes it easier to verify that the correct colors are in use.

Reply:
We usually use the `f` approach, since it's simple and happens to work. The most correct is `(CGFloat)0.08` because the compiler will keep the full 64 bits of precision in 64-bit mode. (If you want to play around with the floating-point representation there are lots of [calculators](https://www.h-schmidt.net/FloatConverter/IEEE754.html) available on the web.)

**However** the error between the two methods is irrelevant for color values, which typically only care about 8 bits anyway (the float gets converted to [0,255]. (Even deeper color spaces, like ones with 16-bit channels, won't be able to see this difference.)

Like all these borderline nitpicky issues, my rule is: they're fine if you want to write them, but don't do large refactors/search-and-replaces or insist that other people change their code—it's not worth the engineering time.

Reply:
Awesome, thank you for the explanation!

Reply:
I occasionally get to use that PhD in computer graphics. 😉